crates/node/engine/src/task_queue/tasks/forkchoice/task.rs

```diff
@@ -49,24 +103,64 @@ impl EngineTaskExt for ForkchoiceTask {
         // Send the forkchoice update through the input.
         let forkchoice = state.create_forkchoice_state();
```

Comment:
re: https://github.com/op-rs/kona/pull/2388/files#r2195674080

note that here, it expects that we'll start building on top of the existing unsafe head, rather than the one declared in the payload attributes we've either derived or formed within the sequencer. This can cause trouble.